src/Sentry/Laravel/EventHandler.php

```diff
@@ -601,4 +667,31 @@ private function cleanupScopeForTaskWithinLongRunningProcessWhen(bool $when): vo
 
         SentrySdk::getCurrentHub()->popScope();
     }
+
+    /**
+     * Construct a full URI
+     * 
+     * @param string $url 
+     * @return UriInterface 
+     */
+    private function getFullUri(string $url): UriInterface
+    {
+        return new Uri($url);
+    }
+
+    /**
+     * Construct a partial URI, excluding the authority and the query and fragment parts.
+     * 
+     * @param UriInterface $uri 
+     * @return UriInterface 
+     */
+    private function getPartialUri(UriInterface $uri): UriInterface
+    {
+        return Uri::fromParts([
+            'scheme' => $uri->getScheme(),
+            'host' => $uri->getHost(),
+            'port' => $uri->getPort(),
+            'path' => $uri->getPath(),
+        ]);
+    }
```

Comment:
We need to apply the same logic for HTTP client spans as well.
@stayallive What would be a good place to share this logic?

Reply:
How about a trait for now? And we work on a larger event system refactor in another PR?

Reply:
Sounds great!